CardPayments/src/main/java/com/paypal/android/cardpayments/CardClient.kt

```diff
@@ -160,26 +160,35 @@ class CardClient internal constructor(
 
     private fun handleBrowserSwitchSuccess(browserSwitchResult: BrowserSwitchResult) {
         ApproveOrderMetadata.fromJSON(browserSwitchResult.requestMetadata)?.let { metadata ->
-            CoroutineScope(dispatcher).launch(approveOrderExceptionHandler) {
-                try {
-                    analyticsService.sendAnalyticsEvent(
-                        "card-payments:3ds:get-order-info:succeeded",
-                        metadata.orderId
-                    )
-                    val deepLinkUrl = browserSwitchResult.deepLinkUrl
-                    val result = CardResult(metadata.orderId, deepLinkUrl)
-                    notifyApproveOrderSuccess(result)
-                } catch (error: PayPalSDKError) {
-                    analyticsService.sendAnalyticsEvent(
-                        "card-payments:3ds:get-order-info:failed",
-                        metadata.orderId
-                    )
-                    throw error
-                }
+            try {
+                analyticsService.sendAnalyticsEvent(
+                    "card-payments:3ds:get-order-info:succeeded",
+                    metadata.orderId
+                )
+                val deepLinkUrl = browserSwitchResult.deepLinkUrl
+                val result = parseApproveOrderDeepLink(metadata.orderId, deepLinkUrl)
+                notifyApproveOrderSuccess(result)
+            } catch (error: PayPalSDKError) {
+                analyticsService.sendAnalyticsEvent(
+                    "card-payments:3ds:get-order-info:failed",
+                    metadata.orderId
+                )
+                notifyApproveOrderFailure(error)
             }
         }
     }
 
+    @Throws(PayPalSDKError::class)
+    private fun parseApproveOrderDeepLink(orderId: String, deepLinkUrl: Uri?): CardResult {
+        val error = deepLinkUrl?.getQueryParameter("error")
```

Comment:
Should we also return an error if the deepLinkUrl is null? Looks like currently that scenario will just return a CardResult with null params - not sure what the desired behavior is

Reply:
Good catch I'll throw an error here when that happens. A user could technically clear their storage whenever they want.

Reply:
Actually thinking it through this won't occur on a storage clear. Either way added a test case for it 👍 .